classes/quiz_settings.php

```diff
@@ -262,105 +262,91 @@ private function compute_config() {
+    private function process_seb_template() {
         $template = template::get_record(['id' => $this->get('templateid')]);
         $this->plist = new property_list($template->get('content'));
 
-        // Process quit settings.
-        $this->process_quit_settings();
-
-        // Add the sensible default options to the configuration and exported SEB files.
+        $this->process_quit_password_settings();
+        $this->process_quit_url_from_template_or_config();
         $this->process_required_enforced_settings();
     }
 
     /**
-     * This function processes the elements that have been set via the modform fields.
-     *
-     * It's called during case USE_SEB_TEMPLATE and USE_SEB_CONFIG_MANUALLY.
+     * Case for USE_SEB_UPLOAD_CONFIG. This creates a plist from an uploaded configuration file, then applies the quiz
+     * password settings and some defaults.
      */
-    private function progress_default_standard_settings() {
-        // Process all settings that are boolean.
-        $this->process_bool_settings();
-
-        // Process quit settings.
-        $this->process_quit_settings();
+    private function process_seb_upload_config() {
+        $file = settings_provider::get_module_context_sebconfig_file($this->get('cmid'));
 
-        // Add all the URL filters.
-        $this->process_url_filters();
+        // If file has been uploaded, overwrite existing config.
+        if (!empty($file)) {
+            $this->plist = new property_list($file->get_content());
+        }
 
-        // Add the sensible default options to the configuration and exported SEB files.
+        $this->process_quit_password_settings();
+        $this->process_quit_url_from_template_or_config();
         $this->process_required_enforced_settings();
     }
 
     /**
-     * Sets or updates some sensible default settings, these are the items 'startURL' and 'sendBrowserExamKey'.
+     * Case for USE_SEB_CLIENT_CONFIG. This creates an empty plist to remove the config stored.
      */
-    private function process_required_enforced_settings() {
-        global $CFG;
-
-        $quizurl = new moodle_url($CFG->wwwroot . "/mod/quiz/view.php", ['id' => $this->get('cmid')]);
-
-        $starturl = $this->plist->get_element_value('startURL');
-        if ($starturl) {
-            $this->plist->update_element_value('startURL', $quizurl->out(true));
-        } else {
-            $this->plist->add_element_to_root('startURL', new CFString($quizurl->out(true)));
-        }
-
-        $sendbrowserexamkey = $this->plist->get_element_value('sendBrowserExamKey');
-        if ($sendbrowserexamkey) {
-            $this->plist->update_element_value('sendBrowserExamKey', true);
-        } else {
-            $this->plist->add_element_to_root('sendBrowserExamKey', new CFBoolean(true));
-        }
+    private function process_seb_client_config() {
+        // Just setup an empty plist.
+        $this->plist = new property_list();
```

Comment:
process_config will `$this->set('config', $this->plist->to_xml());` and I thought it would be nice to clean up the database entry

Reply:
but it won't save empty string, right? It will be some broken seb config string?

Reply:
It will save as an empty plist.
```<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0"><dict/></plist>
```

Reply:
ok. Probably we'd need to do some work to get rid of all possibilities to save anything if we don't  need. But this is outside of this PR.